command/plugins_install.go

```diff
@@ -128,3 +200,159 @@ func (c *PluginsInstallCommand) RunContext(buildCtx context.Context, args []stri
 
 	return 0
 }
+
+func (c *PluginsInstallCommand) InstallFromBinary(opts plugingetter.ListInstallationsOptions, plgin *addrs.Plugin, args *PluginsInstallArgs) int {
+	// As with the other commands, we get the last plugin directory as it
+	// has precedence over the others, and is where we'll install the
+	// plugins to.
+	pluginDir := opts.FromFolders[len(opts.FromFolders)-1]
+
+	var err error
+
+	args.PluginPath, err = filepath.Abs(args.PluginPath)
+	if err != nil {
+		return writeDiags(c.Ui, nil, hcl.Diagnostics{&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Failed to transform path",
+			Detail:   fmt.Sprintf("Failed to transform the given path to an absolute one: %s", err),
+		}})
+	}
+
+	s, err := os.Stat(args.PluginPath)
+	if err != nil {
+		return writeDiags(c.Ui, nil, hcl.Diagnostics{&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Unable to find plugin to promote",
+			Detail:   fmt.Sprintf("The plugin %q failed to be opened because of an error: %s", args.PluginName, err),
+		}})
+	}
+
+	if s.IsDir() {
+		return writeDiags(c.Ui, nil, hcl.Diagnostics{&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Plugin to promote cannot be a directory",
+			Detail:   "The packer plugin promote command can only install binaries, not directories",
+		}})
+	}
+
+	describeCmd, err := exec.Command(args.PluginPath, "describe").Output()
+	if err != nil {
+		return writeDiags(c.Ui, nil, hcl.Diagnostics{&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Failed to describe the plugin",
+			Detail:   fmt.Sprintf("Packer failed to run %s describe: %s", args.PluginPath, err),
+		}})
+	}
+
+	var desc plugin.SetDescription
+	if err := json.Unmarshal(describeCmd, &desc); err != nil {
+		return writeDiags(c.Ui, nil, hcl.Diagnostics{&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Failed to decode plugin describe info",
+			Detail:   fmt.Sprintf("'%s describe' produced information that Packer couldn't decode: %s", args.PluginPath, err),
+		}})
+	}
+
+	semver, err := version.NewSemver(desc.Version)
+	if err != nil {
+		return writeDiags(c.Ui, nil, hcl.Diagnostics{&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Invalid version",
+			Detail:   fmt.Sprintf("Plugin's reported version (%q) is not semver-compatible: %s", desc.Version, err),
+		}})
+	}
+	if semver.Prerelease() != "" {
+		return writeDiags(c.Ui, nil, hcl.Diagnostics{&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Invalid version",
+			Detail:   fmt.Sprintf("Packer can only install plugin releases with this command (ex: 1.0.0), the binary's reported version is %q", desc.Version),
```

Comment:
This doesn't solve the problem for folks using dev builds right. 
```suggestion
			Detail:   fmt.Sprintf("Installation of prerelease binaries with this command is not allowed, the binary's reported version is %q", desc.Version),
```

```
~>  packer plugins install --path ~/.packer.d/plugins/packer-plugin-amazon github.com/hashicorp/happycloud
Error: Invalid version

Packer can only install plugin releases with this command (ex: 1.0.0), the
binary's reported version is "1.2.9-dev"
```